### PR TITLE
Egamma Fall17 V2 Photon IDs

### PR DESCRIPTION
**Pull request by guitargeek**
This PR aims to finally integrate the updated MVA and cut based photon IDs. They are activated in MiniAOD so reco changes are expected. The NanoAOD integration will be done separately in NanoAOD development fork this time.

Shout out to @michelif, @Sam-Harper, @lsoffi and @swagata87!

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from RecoEgamma.PhotonIdentification.Identification.mvaPhotonID_tools import *
+#
```
Comment (peruzzim) [inline RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py]: I think that the import of  `os.path` and the definition of `weightFileBaseDir` are missing here

Comment (peruzzim) [reply RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py]: same with `category_cuts` later in the same file

Comment (guitargeek) [reply RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py]: Oh yes, that's absolutely right! And I noticed I messed up with this PR (I forgot https://github.com/cms-sw/cmssw/pull/24131 was merged in 10_3_X), so I have to adapt the config files a bit.

Comment (guitargeek) [reply RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py]: ...actually we are talking about the backport https://github.com/cms-sw/cmssw/pull/25313. Nothing is missing here.

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- New photon ID made available for miniAOD poduction and analyses
- Changes in the miniAOD outputs correspond to the additional cut and MVA based IDs

**Comment by fabiocos** [comment]
@perrotta "Changes in the miniAOD outputs" refers to the 331 differences in the comparisons?

**Comment by perrotta** [comment]
Yes, they do:

```
/build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison
JR results differ 22 all_mini_OldVSNew_RunEGamma2018Awf136p85
JR results differ 19 all_mini_OldVSNew_TTbar14TeV2023D35wf27434p0
JR results differ 22 all_mini_OldVSNew_RunSinglePh2017Bwf136p788
JR results differ 22 all_mini_OldVSNew_RunPhoton2012Bwf4p53
JR results differ 12 all_mini_OldVSNew_ZMM13TeVwf1330p0
JR results differ 15 all_mini_OldVSNew_TTbar14TeV2023D21wf21234p0
JR results differ 14 all_mini_OldVSNew_TTbar13TeVPUppmx2018wf250202p181
JR results differ 17 all_mini_OldVSNew_TTbar13TeV2017wf10024p0
JR results differ 20 all_mini_OldVSNew_TTbar13TeV2018wf10824p0
JR results differ 18 all_mini_OldVSNew_TTbarPUwf25202p0
JR results differ 12 all_mini_OldVSNew_TTbarwf25p0
JR results differ 19 all_OldVSNew_RunJetHT2016EreMINIAODwf136p7611
JR results differ 13 all_mini_OldVSNew_TTbar13TeVPU2017wf10224p0
JR results differ 22 all_mini_OldVSNew_RunSinglePh2016Bwf136p731
JR results differ 18 all_OldVSNew_RunJetHT2017FreMINIAODwf136p8311
JR results differ 19 all_mini_OldVSNew_TTbar13TeV2019wf11624p0
JR results differ 12 all_mini_OldVSNew_ZMM13TeV2017wf10042p0
JR results differ 17 all_mini_OldVSNew_Higgs200ChargedTauswf9p0
JR results differ 18 all_mini_OldVSNew_TTbar14TeV2023D17wf20034p0
SUMMARY Reco comparison results: 331 differences found in the comparisons
```

File: RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py (hunk 1 above)
Comment (fabiocos) [reply RecoEgamma/PhotonIdentification/python/Identification/mvaPhotonID_Fall17_94X_V2_cff.py]: @peruzzim for what I can see, they are all included within mvaPhotonID_tools, and indeed the unit tests using this fragment looks to work properly

**Comment by fabiocos** [comment]
@peruzzim @fgolf I will move forward with the integration of this development, I understand that the propagation to NanoAOD will happen in a second step

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**